src/app.rs

```diff
@@ -292,6 +297,19 @@ where
             states.stop(StateData::new(world, &mut self.data));
         }
 
+        // Read the Trans queue and apply changes.
+        {
```

Comment:
Consider checking len of `TransQueue` before entering this part, and maybe it would be better to move the part of reading `TransQueue` into separate function